.github/workflows/license.yml

```diff
@@ -0,0 +1,81 @@
+name: Graph Compiler License Check
+
+on:
+  pull_request:
+
+permissions: read-all
+
+jobs:
+  license-check:
+    name: License Check
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+      repository-projects: write
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: true
+
+      - name: Get merge base
+        run: |
+          echo "MERGE_BASE=`git merge-base ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }}`" >> $GITHUB_ENV
+
+      - name: Get changed files
+        run: |
+          echo "CHANGED_FILES=`git diff --name-only $MERGE_BASE ${{ github.event.pull_request.head.sha }} | paste -sd,`" >> $GITHUB_ENV
+
+      - name: Perform license check
+        shell: python
+        run: |
+            import os, datetime, sys
+
+            intel_license: list[str] = [
+                'Copyright (C) YYYY Intel Corporation',
+                'Licensed under the Apache License, Version 2.0 (the "License");',
+                'you may not use this file except in compliance with the License.',
+                'You may obtain a copy of the License at',
+                'http://www.apache.org/licenses/LICENSE-2.0',
+                'Unless required by applicable law or agreed to in writing,',
+                'software distributed under the License is distributed on an "AS IS" BASIS,',
+                'WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.',
+                'See the License for the specific language governing permissions',
+                'and limitations under the License.',
+                'SPDX-License-Identifier: Apache-2.0',
+            ]
+
+            llvm_license: list[str] = [
+                'This file is licensed under the Apache License v2.0 with LLVM Exceptions.',
+                'See https://llvm.org/LICENSE.txt for license information.',
+                'SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception',
+            ]
+
+            def check_license(filename: str, license: list[str]):
```

Comment:
oneDNN's script also provides a utility to auto fix the headers. It is useful when a PR is across two years (to update year). 
I think for LLVM, auto fixing headers are also useful, when we introduce many cpp files in a PR, and we need to manually fill-in the file names in the licence. Maybe we can improve the script in another PR.

Reply:
We can write another small script to handle the fixing job in the future, but not in the automatic CI script.